code/datums/status_effects/buffs.dm

```diff
@@ -323,25 +323,116 @@
 /datum/status_effect/fleshmend
 	id = "fleshmend"
 	alert_type = /atom/movable/screen/alert/status_effect/fleshmend
-	duration = 100
+	//Actual healing lasts for 30 seconds
+	duration = 320
+	var/ticks_passed = 0
 
 /datum/status_effect/fleshmend/tick()
+	ticks_passed ++
 	if(owner.on_fire)
 		linked_alert.icon_state = "fleshmend_fire"
 		return
 	else
 		linked_alert.icon_state = "fleshmend"
-	owner.adjustBruteLoss(-2.5, FALSE)
-	owner.adjustFireLoss(-2.5, FALSE)
-	owner.adjustOxyLoss(-2.5)
-	owner.adjustCloneLoss(-2.5)
-	owner.adjustToxLoss(-2.5, FALSE, TRUE)
+	if(ticks_passed < 2)
+		return
+	else if(ticks_passed == 2)
+		to_chat(owner, "<span class=changeling>We begin to repair our tissue damage...</span>")
+	//Heals 2 brute per second, for a total of 60
+	owner.adjustBruteLoss(-2, FALSE, TRUE)
+	//Heals 1 fireloss per second, for a total of 30
+	owner.adjustFireLoss(-1, FALSE, TRUE)
+	//Heals 5 oxyloss per second for a total of 150
+	owner.adjustOxyLoss(-5, FALSE, TRUE)
+	//Heals 0.5 cloneloss per second for a total of 15
+	owner.adjustCloneLoss(-0.5, TRUE, TRUE)
 
 /atom/movable/screen/alert/status_effect/fleshmend
 	name = "Fleshmend"
 	desc = "Our wounds are rapidly healing. <i>This effect is prevented if we are on fire.</i>"
 	icon_state = "fleshmend"
 
+/datum/status_effect/changeling
+	var/datum/antagonist/changeling/ling
+	var/chem_per_tick = 1
+
+/datum/status_effect/changeling/on_apply()
+	ling = is_changeling(owner)
+	if(!ling)
+		return FALSE
+	return TRUE
```

Comment:
```suggestion
	return is_changeling(owner)
```

Reply:
Probably should add should call parent here

Reply:
Parent call does nothing